src/main/java/net/snowflake/client/jdbc/RestRequest.java

```diff
@@ -303,6 +315,21 @@ public static CloseableHttpResponse execute(
         }
 
         retryCount++;
+
+        // Make sure that any authenticator specific info that needs to be
+        // updated get's updated before the next retry. Ex - JWT token
+        if (authTimeout > 0 && elapsedMilliForTransientIssues > authTimeout) {
+          // check if max retry has been reached
+          if (retryCount > getMaxAuthRetryCount()) {
+            logger.debug("Reached max number of auth retries. Aborting");
+            break;
+          }
+          // check if this is a login-request
+          if (String.valueOf(httpRequest.getURI()).contains("login-request")) {
```

Comment:
I am not understanding this if statement. It looks like there's an assumption that because there was an error with the response and the request was a login request, the error must be due to an authenticator request timeout. However, the error could potentially be unrelated. Can there instead be a check to see if the error is actually caused by an expired JWT token?

Reply:
We want to make sure that we don't get the JWT expired token. So we will renew the JWT token for login requests before the JWT expired error happens so we can avoid getting it in the first place.